### PR TITLE
Add the gRPC-based config spec.

### PR DESCRIPTION
**Pull request by danilvpetrov**
#### What change does this introduce?

This change is the first step of relocating config spec definitions from the `github.com/dogmatiq/configkit/api/internal/pb`package.

#### What issues does this relate to?

Partially addresses https://github.com/dogmatiq/configkit/issues/103.

#### Why make this change?

This change is required to relocated the config spec from `github.com/dogmatiq/configkit/api/internal/pb` into a dedicated package within the current repository. This change is among others to arrange and concentrate all spec definitions for application interoperability within a single repository. 

#### Is there anything you are unsure about?
Possibly the directory structure for the config spec might be reviewed.

### CODE REVIEW

**Comment by danilvpetrov** [comment]
@jmalloc, as we decided to leave `.Validate()` methods and tests out of this PR, I think it's ready for your review.

File: identity.proto [hunk 1]
```diff
@@ -0,0 +1,13 @@
+syntax = "proto3";
+package dogma.interop.v1;
+
+option go_package = "github.com/dogmatiq/interopspec";
+
+// Identity is a protocol buffers representation of the configkit.Identity type.
+message Identity {
```
Comment (jmalloc) [inline identity.proto:7]: The `envelopespec` package should use this type instead of its own `Identity` type.

**Review by jmalloc — CHANGES_REQUESTED** [review]

File: identity.proto (hunk 1 above)
Comment (danilvpetrov) [reply identity.proto:7]: Good catch, fixing now...

Comment (danilvpetrov) [reply identity.proto:7]: This should be fixed with https://github.com/dogmatiq/interopspec/pull/2/commits/d687f462d657d51e3279f2b90054641ed1194466

File: envelopespec/envelope.go [hunk 2]
```diff
@@ -3,6 +3,8 @@ package envelopespec
 import (
 	"errors"
 	"fmt"
+
+	interopspec "github.com/dogmatiq/interopspec"
```
Comment (jmalloc) [inline envelopespec/envelope.go:7]:
```suggestion
	"github.com/dogmatiq/interopspec"
```

File: configspec/api.proto [hunk 3]
```diff
@@ -0,0 +1,100 @@
+syntax = "proto3";
+package dogma.interop.v1.config;
+
+option go_package = "github.com/dogmatiq/interopspec/configspec";
+
+import "github.com/dogmatiq/interopspec/identity.proto";
+
+// Config is a service for inspecting the configuration of Dogma applications
+// hosted by the engine on the server.
+service Config {
```
Comment (jmalloc) [inline configspec/api.proto:10]:
```suggestion
// API is a service for inspecting the configuration of Dogma applications
// hosted by the engine on the server.
service API {
```

**Review by jmalloc — APPROVED** [review]

**Pull request merged**